### ISSUE TITLE
Extend support for identities known by the User Agent and Host OS

### ISSUE THREAD

**Issue by HowardWolosky** [WICG/WebID#51]
Many user agents/web browsers allow a user to associate a federated identity with their browser profile, and some host operating systems allow 1+ federated identities to be associated with the user's account and shared with installed applications.

It would be beneficial if WebID allowed for the promotion of those identities known by the UA and host OS to WebID as selectable identities within its login form, without requiring the user to reenter those credentials.

Additionally, if WebID does allow for the promotion of these preexisting identities to be promoted to the login form, it would follow that we should provide the ability for users to choose if new identities that they authenticate with via WebID should be saved back to the UA (if the profile allows for it) or the host OS (if the federated login type is supported).

**Comment by samuelgoto** [comment]
Ha, what an interesting idea! I like it!

Question: does this necessarily presuppose / imply solving the [NASCAR flag problem](https://github.com/WICG/WebID/blob/master/problems.md#the-nascar-flag-problem) or are you suggesting this can be done even in the absence of a cross IDP account chooser (i.e. Host OS accounts available for a specific IDP)?

**Comment by timcappalli** [comment]
NASCAR should only be a fallback. The RP should be able to invoke CredMan with a list of supported IdPs and DID methods. The browser can check its internal account cache, query wallets and query the OS.

**Comment by kenrb** [comment]
`FederatedCredential` in the CredMan API already essentially provides that capability, but has failed to gain adoption (and correspondingly, universal browser implementation). It might be useful to explore why that is the case.

**Comment by asr-enid** [comment]
Agree, RPs will want to have choice and send users to a specific IDP/Auth method. A lot of them are using identity first flows increasingly to guide user to login options.

**Comment by timcappalli** [comment]
@kenrb , NASCAR is the norm for implementors and consumers so there really hasn't been a push to leverage CredMan. With a big change like browser brokering identity, now is the time to leverage this.

**Comment by samuelgoto** [comment]
This is starting to bubble up in a couple of contexts:

- Exposing FedCM in WebViews and In-App-Browsers (e.g. [facebook's](https://engineering.fb.com/2022/09/30/android/launching-a-new-chromium-based-webview-for-android/)) to get accounts from Android native apps
- Getting accounts from Windows

We are starting to look into what this might look like.

**Comment by hlflanagan** [comment]
I propose closing this issue for now. If the original poster, any of the commenters, or someone in the WG or CG wants to advocate for keeping it open and help move it forward, please comment within two weeks of this label being applied.

**Comment by samuelgoto** [comment]
Closing SGTM, but just wanted to be transparent that some of the feature requests here are being addressed via these affordances https://github.com/w3c-fedid/FedCM/issues/50#issuecomment-4961892872

**Comment by ThisIsMissEm** [comment]
There's also #817 which possibly covers some of this more from a cross-device use-case inspired by the DC API
